### DISCUSSION TITLE
argocd local agent

### DISCUSSION THREAD

**Discussion by yuvalgl** [argoproj/argo-cd#19305] [General]
Hi,
im looking for a way to have a single argocd that can manage multiple clusters on multiple clouds and regions without access to each and every kubeapi.

we have multiple clusters on multiple clouds (on-prem, AWS, AZ and GCP) and on multiple regions and would like to use a single argocd instance that is deployed on one of these clusters in order to have full visibility of the entire environment.

is there a way to achieve this and not open all of the kubeapis to a single argocd instance? maybe some local agent pulling data from the main argocd instance? 

Any help in this issue is greatly appreciated!

**Comment by crenshaw-dev** [comment]
@jannfis is working on a tool to do exactly what you're describing: https://github.com/jannfis/argocd-agent
